streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -466,6 +452,9 @@ public void shouldUpgradeFromEosAlphaToEosBeta() throws Exception {
                 verifyCommitted(expectedCommittedResult);
                 expectedUncommittedResult.addAll(expectedCommittedResult);
             } else {
+                waitForRunning(stateTransitions1);
+                waitForRunning(stateTransitions2);
```

Comment:
Why do we need this? If we don't inject an error, we start phase 6 in RUNNING state and it should not rebalance.

Reply:
Good point